### DISCUSSION TITLE
Bug with name form field in HasOne nova field

### DISCUSSION THREAD

**Discussion by ZsarkZ** [laravel/nova-issues#4716] [Q&A] [answered]
- Laravel Version: 9
- Nova Version: 4
- PHP Version: 8
- Database Driver & Version:
- Operating System and Version: #.#
- Browser type and version: #.#
- Reproduction Repository: https://github.com/###/###

### Description:
I use https://github.com/ebess/advanced-nova-media-library on my form and final name in form which send to server looks like this

`__media__[logo][0]: 'image'`

but if  field to be in HasOne relation form, form field name looks like this

`relationModelName[__media__[logo][0]]: 'image'` 

i think its wrong and the name must be 

`relationModelName[__media__][logo][0]: 'image'`

can you help me with this and explain why that is?

**Comment by crynobone** [comment, marked as answer]
`ebes/advanced-nova-media-library` isn't compatible with `HasOne` field, you can disable Inline editing for HasOne field by adding `exceptOnForms()`.

```php
HasOne::make(__('Employee'), 'employee', Employee::class)->exceptOnForms(),
```

**Reply by ZsarkZ** [reply]
any name who has structure 
`Text::make('Test', 'test[]')`
send like 
`relationModelName[test[]]: ''`
that's how it should be?
maybe this right way?
`relationModelName[test][]: ''`

**Reply by crynobone** [reply]
No plan to change this at the moment.

**Reply by ZsarkZ** [reply]
I understood you! Great answer!
